### PR TITLE
[evm] run hardhat tests in CI

### PR DESCRIPTION
**Pull request by vgao1996**
This makes it possible to run hardhat tests in CI. 

Full list of changes:
- Added a script to install hardhat using npm.
- Added a new github workflow: `ci-test / hardhat-tests`
- Disabled a few broken tests

### CODE REVIEW

File: language/evm/hardhat-examples/contracts/FortyTwo.move [hunk 1]
```diff
@@ -12,10 +12,10 @@ module 0x1::FortyTwo {
     }
 
     // TODO: move-to-yul does not support literal string.
-    #[callable(sig=b"forty_two_as_string() returns (string)"), pure]
-    public fun forty_two_as_string(): vector<u8> {
-         b"forty two"
-    }
+    // #[callable(sig=b"forty_two_as_string() returns (string)"), pure]
```
Comment (junkil-park) [inline language/evm/hardhat-examples/contracts/FortyTwo.move]: does this still cause an error?

**Review by junkil-park — APPROVED** [review]
LGTM!

File: language/evm/hardhat-examples/contracts/FortyTwo.move (hunk 1 above)
Comment (vgao1996) [reply language/evm/hardhat-examples/contracts/FortyTwo.move]: Reenabled it already in a recent push. Sorry for the confusion

**Pull request closed without merging**